rebalance/rebalance.py

```diff
@@ -168,16 +174,35 @@ def rebalance(plugin, outgoing_scid, incoming_scid, msatoshi: Millisatoshi = Non
     description = "%s to %s" % (outgoing_scid, incoming_scid)
     invoice = plugin.rpc.invoice(msatoshi, label, description, retry_for + 60)
     payment_hash = invoice['payment_hash']
-    success_msg = ""
-    try:
-        excludes = []
-        # excude all own channels to prevent unwanted shortcuts [out,mid,in]
-        mychannels = plugin.rpc.listchannels(source=my_node_id)['channels']
-        for channel in mychannels:
-            excludes += [channel['short_channel_id'] + '/0', channel['short_channel_id'] + '/1']
 
+    rpc_result = None
+    count = 0
+    excludes = [my_node_id]  # excude all own channels to prevent shortcuts
+    nodes = {}               # here we store erring node counts
+    maxhops = 1              # retry higher when getroute fails
+    msat_factor = 4
```

Comment:
Not sure, Im still trying and fiddlig with numbers, wanted your feedback on the idea itself.

My rationale was to try very short routes first and then go up to 3,4,5 hops while simultaniously increasing this factor, because it increases the likelyhood a channel will not throw `WIRE_TEMPORARY_CHANNEL_FAILURE` because its to imbalanced.

After this has been tried the also should remove this tricks and try default auto routing of i.e. 8 or 10 max hops.

Also we could consider clearing the `nodes` list which is filled with bad performing nodes on the go.

Reply:
Note: the `msat_factor` is used in a `max(maxhops, msat_factor)` function so it increases as we try longer routes.
We can rename the variable to reflect that ;)